src/otx/cli/utils/experiment.py

```diff
@@ -0,0 +1,253 @@
+"""Utils function for experiments."""
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+import logging
+import multiprocessing as mp
+import os
+import time
+from abc import ABC, abstractmethod
+from pathlib import Path
+from statistics import mean
+from typing import Any, Dict, List, Optional, Union, no_type_check
+
+import psutil
+import yaml
+
+try:
+    import pynvml
+except ImportError:
+    pynvml = None
+
+logger = logging.getLogger(__name__)
+GIB = 1024**3
+AVAILABLE_RESOURCE_TYPE = ["cpu", "gpu"]
+
+
+class ResourceTracker:
+    """Class to track resources usage.
+
+    Args:
+        resource_type (str, optional): Which resource to track. Available values are cpu, gpu or all now.
+                                        Defaults to "all".
+        gpu_ids (Optional[str]): GPU indices to record.
+    """
+
+    def __init__(self, resource_type: str = "all", gpu_ids: Optional[str] = None):
+        if resource_type == "all":
+            self._resource_type = AVAILABLE_RESOURCE_TYPE
+        else:
+            self._resource_type = [val for val in resource_type.split(",")]
```

Comment:
Making proper error raise is a good option to users. (i.e. if users use `xpu` or something)

Reply:
You're right. But it seems that line 99 is better place to raise the error. I'll update it.

Reply:
updated at [22ad518](https://github.com/openvinotoolkit/training_extensions/pull/2500/commits/22ad518e43478297aafd7a5c9cf47d7adf1593c4)